arelle/plugin/validate/ESEF_2022/Const.py

```diff
@@ -0,0 +1,349 @@
+'''
+Created on June 6, 2018
+Version 2022 created on September 19, 2022
+
+Filer Guidelines: https://www.esma.europa.eu/sites/default/files/library/esma32-60-254_esef_reporting_manual.pdf
+
+
+
+@author: Workiva
+(c) Copyright 2022 Workiva, All rights reserved.
+'''
+from __future__ import annotations
+import regex as re
+from typing import Any, Callable
+from arelle.ModelValue import qname
+from arelle.XbrlConst import all, notAll, hypercubeDimension, dimensionDomain, domainMember, dimensionDefault, widerNarrower
+
+browserMaxBase64ImageLength = 5242880 # 5MB
+
+esefTaxonomyNamespaceURIs = {
+    "http://xbrl.ifrs.org/taxonomy/20",
+    "http://xbrl.ifrs.org/taxonomy/20",
+    }
+
+disallowedURIsPattern = re.compile(
+    "http://xbrl.ifrs.org/taxonomy/[0-9-]{10}/full_ifrs/full_ifrs-cor_[0-9-]{10}[.]xsd|"
+    "http://www.esma.europa.eu/taxonomy/[0-9-]{10}/esef_all.xsd"
+    )
+
+
+DefaultDimensionLinkroles = ("http://www.esma.europa.eu/xbrl/role/cor/ifrs-dim_role-990000",)
+LineItemsNotQualifiedLinkrole = "http://www.esma.europa.eu/xbrl/role/cor/esef_role-999999"
+
+qnDomainItemTypes = {qname("{http://www.xbrl.org/dtr/type/non-numeric}nonnum:domainItemType"),
+                     qname("{http://www.xbrl.org/dtr/type/2020-01-21}nonnum:domainItemType")}
+
+
+linkbaseRefTypes = {
+    "http://www.xbrl.org/2003/role/calculationLinkbaseRef": "cal",
+    "http://www.xbrl.org/2003/role/definitionLinkbaseRef": "def",
+    "http://www.xbrl.org/2003/role/labelLinkbaseRef": "lab",
+    "http://www.xbrl.org/2003/role/presentationLinkbaseRef": "pre",
+    "http://www.xbrl.org/2003/role/referenceLinkbaseRef": "ref"
+    }
+
+filenamePatterns = {
+    "cal": "{base}-{date}_cal.xml",
+    "def": "{base}-{date}_def.xml",
+    "lab": "{base}-{date}_lab-{lang}.xml",
+    "pre": "{base}-{date}_pre.xml",
+    "ref": "{base}-{date}_ref.xml"
+    }
+
+filenameRegexes = {
```

Comment:
I would suggest only duplicating the files that you needed to change between versions. The constant files seem fine to share

Reply:
Sharing code between plugins is murky. Right now the priority is to get this merged into the master branch of Arelle. Once it's in we'll work on folding it into the existing ESEF plugin.